### ISSUE TITLE
Build broken by TryFrom on latest nightly 

### ISSUE THREAD

**Issue by konstin** [PyO3/pyo3#136]
With the latest nightly, the build fails with a bunch of errors in `TryFrom` like the following one.

```
note: candidate #1 is defined in an impl of the trait `conversion::PyTryFrom` for the type `objects::sequence::PySequence`
   --> pyo3/src/objects/sequence.rs:252:5
    |
252 |     fn try_from(value: &PyObjectRef) -> Result<&PySequence, Self::Error> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: candidate #2 is defined in an impl of the trait `std::convert::TryFrom` for the type `_`

error[E0034]: multiple applicable items in scope
  --> pyo3/src/objects/stringutils.rs:60:9
   |
60 |         PyString::try_from(ob)?.to_string()
   |         ^^^^^^^^^^^^^^^^^^ multiple `try_from` found
   |
```

This is most likely caused by the stabilization of TryFrom in rust-lang/rust#49305

**Comment by fafhrd91** [comment]
fixes by #137

**Comment by konstin** [comment]
Thanks!
